src/System.Runtime/tests/System/Array.cs

```diff
@@ -1243,6 +1295,16 @@ private static bool ArraysAreEqual<T>(T[] a, T[] b, IComparer<T> comparer)
     }
 
     [Fact]
+    public static void TestSort_Invalid()
+    {
+        IComparer<int> icomparer = new IntegerComparer();
+        Assert.Throws<ArgumentNullException>(() => Array.Sort(null, 0, 0, icomparer));
+        Assert.Throws<ArgumentOutOfRangeException>(() => Array.Sort(new int[] { 1 }, -1, 1, icomparer));
+        Assert.Throws<ArgumentOutOfRangeException>(() => Array.Sort(new int[] { 1 }, 1, -1, icomparer));
+        Assert.Throws<ArgumentException>(() => Array.Sort(new int[] { 1 }, 1, 1, icomparer));
```

Comment:
Yes, there really should be. Our coverage for the array class is really quite bad (~60%) and most of the missing lines are from no input testing.

Since this PR only covers tests that <b>successfully</b> passed in ToF but aren't present in Open, the added tests are somewhat... sporadic. I'll do a look through this PR and clean up the added tests to cover the other overloads and theory-ize them, but at some point we're going to need to add some new tests.